tests/test_device.py

```diff
@@ -174,6 +213,37 @@ def test_check_validity_on_valid_queue(self, mock_device_supporting_paulis):
         # Raises an error if queue or observables are invalid
         mock_device_supporting_paulis.check_validity(queue, observables)
 
+    def test_check_validity_on_valid_queue_with_inverses(self, mock_device_supporting_paulis_and_inverse):
+        """Tests the function Device.check_validity with valid queue and the inverse of operations"""
+        queue = [
+            qml.PauliX(wires=0).inv(),
+            qml.PauliY(wires=1).inv(),
+            qml.PauliZ(wires=2).inv(),
+
+            qml.PauliX(wires=0).inv().inv(),
+            qml.PauliY(wires=1).inv().inv(),
+            qml.PauliZ(wires=2).inv().inv(),
+        ]
+
+        observables = [qml.expval(qml.PauliZ(0))]
+
+        # Raises an error if queue or observables are invalid
+        mock_device_supporting_paulis_and_inverse.check_validity(queue, observables)
```

Comment:
It's a bit strange not having an assert statement here :thinking: Does this function return True?

Reply:
It returns ``None`` implicitly (there is no ``return`` statement in the method). If there is a problem with the validity of the operations and observables, an error is raised.

Reply:
Yeah, in my head it makes sense 😆 It just felt weird not seeing an assert statement!